File: Programmers/level3/level3_네트워크.py
from collections import deque
def solution(n, computers):
    answer = 0
    
    def BFS(q):
        while q:
            i = q.popleft()
            for j, com in enumerate(computers[i]):
                if com and computers[j][j]:
                    q.append(j)
                    computers[j][j] = 0
        
    for i in range(n):
        if computers[i][i]:
            q = deque()
            q.append(i)
            computers[i][i] = 0
            BFS(q)
            answer += 1
    return answer

File: Programmers/level3/level3_네트워크.java
import java.util.*;

class Solution {
    static Queue<Integer> q = new LinkedList<>();
    
    public static void BFS(int n, int[][] computers) {
        while (!q.isEmpty()) {
            int curr = q.poll();
            for (int i = 0; i < n; i++) {
                if (computers[curr][i] == 1 && computers[i][i] == 1) {
                    q.offer(i);
                    computers[i][i] = 0;
                }
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        for (int i = 0; i < n; i++) {
            if (computers[i][i] == 1) {
                q.offer(i);
                computers[i][i] = 0;
                BFS(n, computers);
                answer++;
            }
        }
        return answer;
    }
}

